### PR TITLE
docs: Add example about retriever

### PR DESCRIPTION
**Pull request by devalexandre**
A example about retriever, how use and why need use

### PR Checklist

- [x] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [x] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [x] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [ ] Describes the source of new concepts.
- [ ] References existing implementations as appropriate.
- [ ] Contains test coverage for new functions.
- [ ] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

### CODE REVIEW

File: docs/docs/modules/data_connection/retrievers/index.mdx [hunk 1]
```diff
@@ -10,5 +10,76 @@ import DocCardList from "@theme/DocCardList";
...
+			log.Fatal(err)
+		}
+	}   
+```
+
+Now Create a retriever
+```go
+    searchQuery := "how to make a cake"
+ 	optionsVector := []vectorstores.Option{
+		vectorstores.WithScoreThreshold(0.80),
```
Comment (tmc) [inline docs/docs/modules/data_connection/retrievers/index.mdx:71]: it'd be good to include some in-line comments about what parameters mean / influence.

**Review by tmc — COMMENTED** [review]
Thanks for adding this, a small comment that i think would improve readability/interpretability.

**Comment by tmc** [comment]
I'm happy to accept this change but will hold off for a bit so you have an opportunity to add a bit more detail as requested

File: docs/docs/modules/data_connection/retrievers/index.mdx [hunk 2]
```diff
@@ -10,5 +10,76 @@ import DocCardList from "@theme/DocCardList";
...
+		log.Fatal(err)
+	}
+
+	embedder, err := embeddings.NewEmbedder(llm)
+	if err != nil {
+		log.Fatal(err)
+	}
+```
+
+After it chose a storage vector like pinecone, postgres, Qdrant, in example I'll use qdraant
```
Comment (corani) [inline docs/docs/modules/data_connection/retrievers/index.mdx:43]: typo:
```suggestion
After it chose a storage vector like pinecone, postgres, Qdrant, in example I'll use qdrant
```

and you might want to check the capitalization of these product names, e.g. "PostgreSQL"

**Review by tmc — APPROVED** [review]
LGTM

**Pull request merged**